rclc/include/rclc/type_support.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLC__TYPE_SUPPORT_H_
+#define RCLC__TYPE_SUPPORT_H_
+
+#include <stddef.h>
+
+#include "rosidl_generator_c/message_type_support_struct.h"
+
+typedef struct rclc_type_support_t rclc_type_support_t;
+struct rclc_type_support_t
```

Comment:
maybe `rclc_message_type_support_t` would be more explicit as (I guess) we will need to create a service equivalent in the future.